pydantic_ai_slim/pydantic_ai/builtin_tools.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations as _annotations
+
+from abc import ABC
+from dataclasses import dataclass, field
+from typing import Literal
+
+from typing_extensions import TypedDict
+
+__all__ = ('AbstractBuiltinTool', 'WebSearchTool', 'UserLocation')
+
+
+class AbstractBuiltinTool(ABC):
+    """A builtin tool that can be used by an agent.
+
+    This class is abstract and cannot be instantiated directly.
+    """
+
+
+class UserLocation(TypedDict, total=False):
```

Comment:
It's easier to handle this in the models if it's a `TypedDict`, since it matches the type.